plugins/org.python.pydev/pysrc/pydevconsole.py

```diff
@@ -508,6 +508,17 @@ def console_exec(thread_id, frame_id, expression, dbg):
     port, client_port = sys.argv[1:3]
     from _pydev_bundle import pydev_localhost
 
+    # Automatically tear down the console if the parent Eclipse goes away
+    # https://www.brainwy.com/tracker/PyDev/860
+    import time
+    def exit_on_parent_death():
+        while True:
+            time.sleep(5)
+            # http://stackoverflow.com/questions/269494/how-can-i-cause-a-child-process-to-exit-when-the-parent-does
+            if os.getppid() == 1:
```

Comment:
os.getppid() isn't available on windows, so, this only works on Linux -- at least on python 2.7 (so, this code needs to check that it does exist, otherwise there's no point in creating the thread).

Reply:
Ok thanks - will take a look. Do you know if this problem exists on windows as well or is child proc killing handled differently? I’ve not got easy access to Windows to test...

Reply:
The problem also exists on windows (just tested), but on newer versions of python `os.getppid` is available on windows too, so, fixing just when `os.getppid` is available seems good.

Reply:
I think parentpid => 1 is OS dependent.  On Windows at least it seems that the parent pid isn't reset and can just belong to another process:
https://stackoverflow.com/a/185298

I think it would probably be better to terminate the pydevconsole on socket disconnect.  I'm trying to figure out the best way for debugging the console scripts.  At the moment Pydev prefix the installation pysrc directory to the path (rather than the workspace one):
 
![image](https://user-images.githubusercontent.com/55716/31823640-abba890e-b5a4-11e7-9183-a0a947481b33.png)

Do you have a trick for running these tests in the IDE?

Reply:
Try what I might I can't get the pydev debugger to stop in pydevconsole.py - even when running remotely and attaching the debugger to it with `import pydevd; pydevd.settrace()` ...